src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -284,91 +275,149 @@ runTestCase( const std::shared_ptr<xrt_core::device>& _dev, const std::string& p
-        _ptTest.put("status", test_token_failed);
-      }
-    } catch (const std::exception& e) {
-      logger(_ptTest, "Error", e.what());
+      return true;
+    } else {
+      logger(_ptTest, "Error", os_stdout.str());
+      logger(_ptTest, "Error", os_stderr.str());
       _ptTest.put("status", test_token_failed);
+      return false;
     }
+  } catch (const std::exception& e) {
+    logger(_ptTest, "Error", e.what());
+    _ptTest.put("status", test_token_failed);
+    return false;
   }
-  else {
-    //check if testcase is present
-    std::string xrtTestCasePath = "/opt/xilinx/xrt/test/" + py;
-    boost::filesystem::path xrt_path(xrtTestCasePath);
-    if (!boost::filesystem::exists(xrt_path)) {
-      logger(_ptTest, "Error", boost::str(boost::format("Failed to find %s") % xrtTestCasePath));
-      logger(_ptTest, "Error", "Please check if the platform package is installed correctly");
-      _ptTest.put("status", test_token_failed);
-      return;
-    }
-    // log testcase path for debugging purposes
-    logger(_ptTest, "Testcase", xrtTestCasePath);
+  return false;
+}
 
-    std::vector<std::string> args = { "-k", xclbinPath,
-                                      "-d", xrt_core::query::pcie_bdf::to_string(xrt_core::device_query<xrt_core::query::pcie_bdf>(_dev)) };
-    int exit_code;
-    try {
-      if (py.find(".exe") != std::string::npos)
-        exit_code = XBU::runScript("", xrtTestCasePath, args, "Running Test", MAX_TEST_DURATION, os_stdout, os_stderr);
-      else
-        exit_code = XBU::runScript("python", xrtTestCasePath, args, "Running Test", MAX_TEST_DURATION, os_stdout, os_stderr);
+/*
+ * Helper function to query support for black-box tests or run them.
+ */
+bool
+runBlackBox(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptree& _ptTest,
+            std::ostringstream & os_stdout, std::ostringstream & os_stderr, bool check_supported)
+{
+  std::string xclbinPath;
+  try {
+    xclbinPath = find_xclbin_path(_dev, _ptTest, _ptTest.get<std::string>("xclbin"));
+  } catch (...) {
+    return false;
+  }
 
-      if (exit_code == EOPNOTSUPP) {
-        _ptTest.put("status", test_token_skipped);
-      }
-      else if (exit_code == EXIT_SUCCESS) {
-        _ptTest.put("status", test_token_passed);
-      }
-      else {
-        logger(_ptTest, "Error", os_stdout.str());
-        logger(_ptTest, "Error", os_stderr.str());
-        _ptTest.put("status", test_token_failed);
-      }
-    } catch (const std::exception& e) {
-      logger(_ptTest, "Error", e.what());
-      _ptTest.put("status", test_token_failed);
-    }
+  boost::filesystem::path xclbin_path(xclbinPath);
+  auto test_dir = xclbin_path.parent_path().string();
+  if (check_supported)
+    logger(_ptTest, "Xclbin", test_dir);
+
+  bool json_exists = boost::filesystem::exists(test_dir + "/platform.json") ? true : false;
+
+  // Need json_exists to check if test is supported.
+  if (check_supported && !json_exists) {
```

Comment:
If the path object is produced early, then the boost::filesystem::exists() method should be used here.